hail/python/hail/vds/methods.py

```diff
@@ -443,6 +443,101 @@ def filter_variants(vds: 'VariantDataset', variants_table: 'Table', *, keep: boo
     return VariantDataset(vds.reference_data, variant_data)
 
 
+@typecheck(vds=VariantDataset,
+           intervals=expr_array(expr_interval(expr_any)),
+           keep=bool,
+           mode=enumeration('variants_only', 'split_at_boundaries', 'unchecked_filter_both'))
+def _parameterized_filter_intervals(vds: 'VariantDataset',
```

Comment:
What is the point of having this be a function at all? It's hidden from the public API and used in one place. Furthermore, the modes are short and seem like they can be inlined where necessary.

Reply:
ah, you've pointed out that I failed to call this from `hl.vds.filter_intervals`!